### ISSUE TITLE
[NEW RULE] Check "repository.url" value

### ISSUE THREAD

**Issue by kravetsone** [bluwy/publint#97]
Please add rule which check `repository.url`

```bash
npm warn publish npm auto-corrected some errors in your package.json when publishing.  Please run "npm pkg fix" to address these errors.
npm warn publish errors corrected:
npm warn publish "repository.url" was normalized to "git+https://github.com/kravetsone/esbuild-plugin-autoload.git"
```

Also i wrote the [publint extenstion for VS Code](https://github.com/kravetsone/vscode-publint)

**Comment by bluwy** [comment]
Sounds good to me. I'm not sure if there's many validations for `package.json`, but if there's only a few like this, it seems fine to me. As the current rules are focused on runtime compatibility only.



> Also i wrote the [publint extenstion for VS Code](https://github.com/kravetsone/vscode-publint)

Amazing, thanks for creating it. Going to give it a spin.

**Comment by Namchee** [comment]
Would like to pitch in for this, why don't we validate the whole `repository` field instead?

**Comment by bluwy** [comment]
Validating the entire `repository` is fine too 👍 As long as they're kept under a single rule name, maybe "INVALID_REPOSITORY_VALUE", and the message changes based on what's wrong. Doing that might be a bit tricky though but the next WIP version should help. At the meantime if we want to implement this now, a generic warning message is also fine.

About `eslint-plugin-package-json`, there's certainly overlap and I think that's ok too, as long as this tool focuses on linting stuff related to publishing packages and runtime compatibility.

**Comment by tunnckoCore** [comment]
@bluwy @Namchee it's important to also check case-sensitivity, because it turn out it's case-sensitive! Found several issues addressing this. I couldn't believe but it's fact. And when i think about it, it makes sense because everything is HASH checksumed to the max.
